cfme/infrastructure/virtual_machines.py

```diff
@@ -512,6 +513,24 @@ def is_displayed(self):
         return self.in_infra_vms and self.title.text == expected_title
 
 
+class InfraVmCompareToolbar(View):
+    """The toolbar on the compare page"""
+    all = Button(title='All Attributes')
+    different = Button(title='Attributes with different values')
+    same = Button(title='Attributes with same values')
+
+
+class InfraVmCompareView(CompareView):
+    @property
+    def is_displayed(self):
+        title = "Compare VM or Template"
+        return (self.logged_in_as_current_user and
```

Comment:
`self.logged_in_as_current_user` is taken care of by `BaseLoggedInPage`, so we don't need to check that here.